src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -160,16 +160,45 @@ private void MoveUnprocessedFilesBackToCache()
 
             foreach (var filePath in Directory.EnumerateFiles(_processingDirectoryPath))
             {
-                try
-                {
-                    var destinationPath = Path.Combine(_isolatedCacheDirectoryPath, Path.GetFileName(filePath));
-                    _options.LogDebug("Moving unprocessed file back to cache: {0} to {1}.",
-                        filePath, destinationPath);
-                    File.Move(filePath, destinationPath);
-                }
-                catch (Exception e)
+                var destinationPath = Path.Combine(_isolatedCacheDirectoryPath, Path.GetFileName(filePath));
+                _options.LogDebug("Moving unprocessed file back to cache: {0} to {1}.", filePath, destinationPath);
+
+                const int maxAttempts = 3;
+                for (var attempt = 1; attempt <= maxAttempts; attempt++)
                 {
-                    _options.LogError("Failed to move unprocessed file back to cache: {0}", e, filePath);
+                    try
+                    {
+                        File.Move(filePath, destinationPath);
+                        break;
+                    }
+                    catch (Exception ex)
+                    {
+                        if (!File.Exists(filePath))
+                        {
+                            _options.LogDebug(
+                                "Failed to move unprocessed file back to cache (attempt {0}), " +
```

Comment:
is it worth including the exception in the log message?

Reply:
Probably not until the actual failure